distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestCatalog.java

```diff
@@ -2962,14 +2962,19 @@ public void testUpdateStorageCannotOverrideResourceUri()
                 .post(REST_PATH.getUrl())
                 .getHeader("id");
 
-        given().multiPart("parse.resource", tmpFile)
-                .multiPart(Core.RESOURCE_URI, overrideResourceUri)
-                .expect()
-                .log()
-                .headers()
-                .statusCode(HttpStatus.SC_BAD_REQUEST)
-                .when()
-                .put(REST_PATH.getUrl() + id);
+        expect("Sending bad HTTP PUT request with overridden resource-uri").within(1,
+                TimeUnit.MINUTES)
+                .until(() -> {
+                    Response response = given().multiPart("parse.resource", tmpFile)
+                            .multiPart(Core.RESOURCE_URI, overrideResourceUri)
+                            .expect()
+                            .log()
+                            .headers()
+                            .when()
+                            .put(REST_PATH.getUrl() + id);
```

Comment:
❗️ Instead of trying the PUT multiple times, it might be safer to do a GET until we know the metacard with the right resource-uri is available and then make one PUT request and check the status code. Calling PUT multiple times assumes that PUT will only fail if the resource-uri has changed which might not alway be the case.